OpenRA.Mods.Common/Traits/BotModules/Squads/States/AirStates.cs

```diff
@@ -251,9 +209,10 @@ public void Tick(Squad owner)
 
 			foreach (var a in owner.Units)
 			{
-				if (!ReloadsAutomatically(a) && !FullAmmo(a))
+				var ammoPools = a.TraitsImplementing<AmmoPool>();
```

Comment:
This has the same problem.

Fwiw, I'm still not a fan of the approach, but if you both think it is worth it we can keep it.

Reply:
Ummm, I quite don't know how many performance it will bring. But AI tick can sometimes lag the game (mainly pathfinding, though).

Reply:
should be fixed